crates/service/tests/router_test.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023-, Edge & Node, GraphOps, and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::{Duration, SystemTime};
+
+use alloy::primitives::Address;
+use axum::http::Request;
+use axum_extra::headers::Header;
+use indexer_config::{BlockchainConfig, GraphNodeConfig, IndexerConfig, NonZeroGRT};
+use indexer_monitor::EscrowAccounts;
+use indexer_service_rs::{
+    service::{ServiceRouter, TapReceipt},
+    QueryBody,
+};
+use reqwest::{StatusCode, Url};
+use sqlx::PgPool;
+use test_assets::{create_signed_receipt, INDEXER_ALLOCATIONS, TAP_EIP712_DOMAIN};
+use tokio::sync::watch;
+use tower::ServiceExt;
+use wiremock::{
+    matchers::{method, path},
+    Mock, MockServer, ResponseTemplate,
+};
+
+#[sqlx::test(migrations = "../../migrations")]
+async fn full_integration_test(database: PgPool) {
+    let http_client = reqwest::Client::builder()
+        .tcp_nodelay(true)
+        .build()
+        .expect("Failed to init HTTP client");
+
+    let allocation = INDEXER_ALLOCATIONS
+        .values()
+        .collect::<Vec<_>>()
+        .pop()
+        .unwrap()
+        .clone();
```

Comment:
Is there a reason to collect instead of `let allocation = INDEXER_ALLOCATIONS.values().next().unwrap().clone();`?